cppapi/client/devapi_attr.cpp

```diff
@@ -563,6 +563,7 @@ DeviceAttribute::DeviceAttribute(string &new_name, short datum):ext(new DeviceAt
 	ShortSeq = new(DevVarShortArray);
 	ShortSeq->length(1);
 	ShortSeq[0] = datum;
+	data_type = DEV_SHORT;
```

Comment:
Looks to me that the whole constructor can be delegated to the *DeviceAttribute(const cha*)*:

```c++
DeviceAttribute::DeviceAttribute(string &new_name, short datum):ext(new DeviceAttributeExt)
{
    this(new_name.c_str(), datum);
}
```

Is this available in C++? Otherwise a helper method may be introduced.

Reply:
@Ingvord You can do that in C++11 only.

And there it must be writtten as

```cpp
DeviceAttribute::DeviceAttribute(string &new_name, short datum)
:
ext(new DeviceAttributeExt),
DeviceAttribute(new_name.c_str(), datum)
{}
```

Reply:
@t-b thanks for the hint!

Reply:
Indeed, this works only in C++11. So I think we could do that for Tango 10.
In any case, I agree with Igor we could refactor using a method which would initialize most of the members of this class.
There is a lot of code duplication in this file in all the constructors.
I will propose another Pull Request for this refactoring in order to avoid to pollute this one.
I will propose it just after this PR has been merged.